modules/csharp/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,227 @@
+= Working with Vector Search
+:page-status: Beta
+:page-edition: Enterprise
+:page-aliases: 
+ifdef::show_edition[:page-edition: {release}]
+ifdef::prerelease[:page-status: {prerelease}]
+:page-role:
+:description: Use Vector Search with Full Text Search and Query.
+:keywords: edge AI api swift ios macos apple vector search generative
+
+[abstract]
+{description}
+
+== Use Vector Search
+
+To configure a project to use vector search, follow the xref:csharp:gs-install.adoc[installation instructions] to add the Vector Search extension.
+
+NOTE: You must install Couchbase Lite to use the Vector Search extension.
+
+== Create a Vector Index
+
+This method shows how you can create the default vector index configuration using the Couchbase Lite Vector Search extension.
+
+[source, csharp]
+----
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-default-config]
+
+----
+
+First, we initialize the `config` object with a new `VectorIndexConfiguration` object initialized with the following parameters:
+
+* The expression of the data as a vector.
+
+* The width or `dimensions` of the vector index is set to `3`.
+
+* The amount of `centroids` is set to `2`.
+This means that there will be two buckets with a single centroid each that gathers together similar vectors.
```

Comment:
This results in two buckets, each with a single centroid that acts to gather similar vectors together within that bucket.

? not sure I'm getting centroids...!